pkg/workflows/upgrade.go

```diff
@@ -119,37 +120,50 @@ func (s *setupAndValidateTasks) Run(ctx context.Context, commandContext *task.Co
 		commandContext.SetError(err)
 		return nil
 	}
+	currentSpec, err := commandContext.ClusterManager.GetCurrentClusterSpec(ctx, commandContext.ManagementCluster, commandContext.ClusterSpec.Cluster.Name)
+	if err != nil {
+		commandContext.SetError(err)
+		return &CollectDiagnosticsTask{}
+	}
+	commandContext.CurrentClusterSpec = currentSpec
+	s.FirstRunComplete = true
 
 	return &updateSecrets{}
 }
 
 func (s *setupAndValidateTasks) validations(ctx context.Context, commandContext *task.CommandContext) []validations.Validation {
-	return []validations.Validation{
+	result := []validations.Validation{
 		func() *validations.ValidationResult {
 			return &validations.ValidationResult{
 				Name: fmt.Sprintf("%s Provider setup is valid", commandContext.Provider.Name()),
 				Err:  commandContext.Provider.SetupAndValidateUpgradeCluster(ctx, commandContext.ManagementCluster, commandContext.ClusterSpec),
 			}
 		},
-		func() *validations.ValidationResult {
+	}
+	if !s.FirstRunComplete {
```

Comment:
This was put into place because some preflight validations could fail depending on where the CLI failed & where we are attempting to resume it. A future addition of this feature is to add more validations to catch instances where the user changed a field between runs